### PR TITLE
Ol testing cleanup

### PR DESCRIPTION
**Pull request by will-moore**
This re-activates the units tests. We now use the ```karma-webpack``` plugin to pre-process the tests, building a bundle for each to serve to the HeadlessChrome.

Should see all tests passing in the travis log.
I removed the ```suite.html``` file which allowed running the ```mocha``` tests directly in the browser without ```karma``` since we rely on ```karma-webpack``` to build each test (include the src code dependencies) first.

Also updated docs at https://github.com/will-moore/omero-iviewer/blob/ol_testing_cleanup/test/README.rst

NB: this PR is on top of #223

### CODE REVIEW

**Comment by will-moore** [comment]
After that last commit, I finally have 1 placeholder unit test running:
```
     [exec] HeadlessChrome 71.0.3578 (Linux 0.0.0): Executed 2 of 2[32m SUCCESS[39m (0.01 secs / 0.001 secs)
     [exec] [32mTOTAL: 2 SUCCESS[39m
```

**Comment by will-moore** [comment]
For some reason travis has started failing.
It originally passed for c9eb8b4 above, then started failing when running the tests.
I then pushed c9eb8b4 to another PR #226 to re-run the test at that point and it failed (which is why that commit now shows fail).
The error is not very clear:

```
     [echo] Running Test Suite ....
     [exec] /omero-iviewer/node_modules/karma/lib/server.js:118
     [exec]   async start () {
     [exec]         ^^^^^
     [exec] 
     [exec] SyntaxError: Unexpected identifier
     [exec]     at createScript (vm.js:56:10)
     [exec]     at Object.runInThisContext (vm.js:97:10)
     [exec]     at Module._compile (module.js:549:28)
     [exec]     at Object.Module._extensions..js (module.js:586:10)
     [exec]     at Module.load (module.js:494:32)
     [exec]     at tryModuleLoad (module.js:453:12)
     [exec]     at Function.Module._load (module.js:445:3)
     [exec]     at Module.require (module.js:504:17)
     [exec]     at require (internal/module.js:20:19)
     [exec]     at Object.<anonymous> (/omero-iviewer/node_modules/karma/lib/cli.js:8:16)
     [exec] Result: 1
```

**Comment by will-moore** [comment]
Tests are passing for me locally with ```ant unit-tests``` but not via local ```omero-infra``` testing (same error as above). Don't know what changed - not node_modules (deleting and re-installing via ```ant unit-tests``` works OK). Any ideas @jburel ?

File: test/unit/regions.js [hunk 1]
```diff
@@ -71,126 +86,118 @@ describe("Regions", function() {
...
+        assert.instanceOf(feature.getGeometry(), Line);
         expect(feature.getGeometry().getFlatCoordinates()).to.eql(
             [4897,-2756,4885,-2786,4826,-2904]);
 
-        feature = ome.ol3.utils.Regions.featureFactory(polygon_info);
-        assert.instanceOf(feature, ol.Feature);
-        assert.instanceOf(feature.getGeometry(),  ol.geom.Polygon);
+        feature = featureFactory(polygon_info);
+        assert.instanceOf(feature, Feature);
+        assert.instanceOf(feature.getGeometry(), OlPolygon);
```
[the remark this answers is not in the document]
Comment (will-moore) [reply test/unit/regions.js]: Probably makes sense, although that's not what was there before (possible typo). Polygon is a subclass of ```ol.geom.Polygon (OlPolygon)``` anyway.

**Comment by will-moore** [comment]
@jburel How would I get to a terminal in the ```test-infra``` docker so that I could try manually running ```ant unit-tests``` or ```karma start --single-run --browsers ChromeHeadless``` there?

**Comment by will-moore** [comment]
@jburel - I think this could be due to the version of node used (https://stackoverflow.com/questions/47434510/unexpected-token-async/47439174).
Do you know what version of node is used by omero-test-infra? I can't see where this could be specified?

**Comment by sbesson** [comment]
@will-moore I think node is not installed by `omero-test-infra` by default but rather in https://github.com/ome/omero-iviewer/blob/master/.omeroci/app-deps. In that case, the version seems to be the one that is shippped by the base CentOS packages?

**Comment by will-moore** [comment]
Thanks - the strange thing is that this was working fine until it wasn't (same commit initially was passing and then started failing without any change here or in omero-test-infra).

**Pull request merged**